packages/compass-smoke-tests/src/cli.ts

```diff
@@ -198,38 +200,69 @@ async function run() {
     ])
   );
 
-  const { kind, appName, filepath, autoUpdatable } = await getTestSubject(
-    context
-  );
-  const install = getInstaller(kind);
+  const {
+    kind,
+    appName,
+    filepath,
+    buildInfo: { channel, version },
+    autoUpdatable,
+  } = await getTestSubject(context);
 
   try {
     if (context.tests.length === 0) {
       console.log('Warning: not performing any tests. Pass --tests.');
     }
 
     for (const testName of context.tests) {
+      const installerPath =
+        testName === 'auto-update-to'
+          ? await getLatestRelease(
+              channel,
+              context.arch,
+              kind,
+              context.forceDownload
+            )
+          : filepath;
```

Comment:
Won't this depend on the order if multiple `tests` are passed to the CLI?
I suggest we make this a bit less powerful and run only one test per invocation of the CLI (effectively changing `context.tests` to `context.test`) to make it easier to reason about what installer needs to run first.

Reply:
Looking at this again, it does feel a bit misplaced here, instead of being a part of the actual test function, which would increase the cohesion 🤔

Reply:
OK what I've done now is move all this stuff into the tests. I also changed it so that auto-update-to does not unnecessarily download the newly packaged app when it is going to be installing the latest release anyway.

Let me know what you think.